Cargo.toml

```diff
@@ -29,12 +29,17 @@ protobuf = "2"
 slog = "2.2"
 quick-error = "1.2.2"
 raft-proto = { path = "proto", version = "0.6.0-alpha", default-features = false }
-rand = "0.7.0"
+# Stick with 0.6 (rather than 0.7) for now, since most of the ecosystem has not
+# upgraded yet and this saves duplicate deps.
+rand = "0.6.5"
 hashbrown = "0.5"
 fail = { version = "0.3", optional = true }
 getset = "0.0.7"
 slog-stdlog = "4"
 slog-envlogger = "2.1.0"
+# We don't actually use this crate, but by pinning to 0.5.1 it forces Cargo to
+# avoid 0.5.2 and thus pulling in a transitive dep with a security vulnerability.
```

Comment:
If we don't use this crate, why pinning it here? Doesn't it introduce extra dependency?

Reply:
No. It is a transitive dependency already so it is always depended on. If we don't pin it, then Cargo will select the more recent 0.5.2, and that has a security vulnerability. Note that in TiKV we stay on 0.5.1 because of the lock file, so its not an issue there.

Reply:
I think it's a bad idea to solve it in raft library. It should be the application's responsibility to choose a good version. Besides, it also prevent downstream from choosing a good version like 0.5.3 or 0.4.0.

Reply:
I agree it is not ideal, but it seems better than not doing it and getting a vulnerable dep (note that this is currently a problem, so this change only makes things better). I don't mind taking this out though if you prefer.